From 05dc9f76fc54dc4c594c5b239a171b02dae6dd80 Mon Sep 17 00:00:00 2001
From: Neo Fang <neo.fang@intel.com>
Date: Thu, 14 Sep 2023 07:51:44 +0000
Subject: [PATCH] Enable v4l2loopback as part of ivi image

Issues: No V4L2loopback driver and device node when system
boots up.

RootCause: V4L2loopback is not enabled in celadon ivi kernel.

Fix Details: Enable V4L2loopback driver module as part of
celadon kernel, create device node /dev/video50 when system
boots up.

Tracked-On: OAM-112223
Signed-off-by: Neo Fang <neo.fang@intel.com>
---
 celadon_ivi/device.mk                         |   3 +-
 celadon_ivi/init.rc                           |   5 +
 celadon_ivi/mixins.spec                       |   1 +
 celadon_tablet/BoardConfig.mk                 |   3 +
 .../linux-intel-lts2021/x86_64_defconfig      | 150 ++++++++++++------
 27 files changed, 378 insertions(+), 173 deletions(-)

diff --git a/celadon_ivi/config-lts/linux-intel-lts2021/x86_64_defconfig b/celadon_ivi/config-lts/linux-intel-lts2021/x86_64_defconfig
index 7b4928e..d026c7f 100644
--- a/celadon_ivi/config-lts/linux-intel-lts2021/x86_64_defconfig
+++ b/celadon_ivi/config-lts/linux-intel-lts2021/x86_64_defconfig
@@ -3776,6 +3776,7 @@ CONFIG_VIDEO_V4L2_SUBDEV_API=y
 # CONFIG_VIDEO_ADV_DEBUG is not set
 # CONFIG_VIDEO_FIXED_MINOR_RANGES is not set
 # end of Video4Linux options
+CONFIG_VIDEO_V4L2LOOPBACK=m
 
 #
 # Media controller options
diff --git a/celadon_ivi/init.rc b/celadon_ivi/init.rc
index 6a38d90..32f415e 100644
--- a/celadon_ivi/init.rc
+++ b/celadon_ivi/init.rc
@@ -819,4 +819,9 @@ on post-fs
     insmod /vendor/lib/modules/ti960.ko
     insmod /vendor/lib/modules/isx031.ko
     insmod /vendor/lib/modules/intel-ipu6-isys.ko
+##############################################################
+# Source: device/intel/mixins/groups/v4l2loopback/true/init.rc
+##############################################################
+on post-fs
+    insmod /vendor/lib/modules/v4l2loopback.ko devices=1 video_nr=50 card_label="LicCam" exclusive_caps=1 broadcast=1 nlgroup=22
 # ------------------ END MIX-IN DEFINITIONS ------------------
diff --git a/celadon_ivi/mixins.spec b/celadon_ivi/mixins.spec
index fa35735..07192af 100755
--- a/celadon_ivi/mixins.spec
+++ b/celadon_ivi/mixins.spec
@@ -92,3 +92,4 @@ houdini: true
 neuralnetworks: true
 docker: true
 ipu: true
+v4l2loopback: true
-- 
2.17.1

